### ISSUE TITLE
Add Medication component should be linkedd to backend

### ISSUE THREAD

**Issue by manaschubby** [manaschubby/MediLink-Pro#32]
Add Medication functionality to be implemented by connecting the main process using IPC communication. Either new medicine should be created or if pre-existing exists it should directly create a medication for the patient.

**Comment by TreLynnnn** [comment]
I'd love to help out with this project, if possible!

**Comment by manaschubby** [comment]
@TreLynnnn Thank you for reaching out! We would love to have you on board with MediLink Pro.

Clone the repo and start the app on your machine. Add a patient and then click on the patient in the Patient Table to open his/her individual page. You'll find that the Add Appointment and Add Diagnosis pop ups are functional, but Add Medication are not. We need to make that functional too. Using IPC Communication the data needs to be send to the main process and added into the database.

Do you want me to assign the issue to you?

**Comment by TreLynnnn** [comment]
Yes, please! Glad I can help.

**Comment by manaschubby** [comment]
Awesome! I have assigned the issue to you.

When you have finished resolving the issue, please fork the repo and open a Pull request. If possible, describe the changes that you have made to resolve the issue. Request a review from @Shlokkzz or @manaschubby. 

Thank you for contributing.

**Comment by manaschubby** [comment]
Please reach out if you are facing any issues in understanding the code or setting up the development environment on your machine. We are here to help!

**Comment by TreLynnnn** [comment]
When adding a new patient, clicking the “+” opens a new dashboard window but doesn’t allow the user to add the new patient info. Opens a new window each time. — Getting an error of Unhandled Promise rejection. This prevents me from viewing the patient table.

**Comment by manaschubby** [comment]
@TreLynnnn You are right. We are facing this issue because of our migration to HashRouter from BrowserRouter. I shall resolve this issue and update the branch. Till then in the `main.js` file, in the `ipcMain.on("add-patient", (e, arg) => {`  function change the `addPatientWindow.loadURL("http://localhost:3000/add-patient");` to `addPatientWindow.loadURL("http://localhost:3000#/add-patient");`. This should let you open the add-patient window properly.

**Comment by manaschubby** [comment]
Hi @TreLynnnn  Please sync your fork, there are some changes regarding our migration from BrowserRouter to HashRouter. You'll be able to open the AddPatient window properly now

**Comment by manaschubby** [comment]
Was your issue resolved?

**Comment by TreLynnnn** [comment]
I've synced and the window now opens for user to input patient info, but once "add patient" is clicked it loads indefinitely. I'm getting a mongoose timeout error. Should I connect my own personal database key for the time being? [image](https://github.com/manaschubby/MediLink-Pro/assets/106888539/38dd7548-f292-4a0b-a737-cf9b05f065c3)

**Comment by manaschubby** [comment]
Please duplicate the `.env.example` file and rename it to `.env`.

That will connect it to your local mongodb server. If you plan to use Atlas. Add the URI to the `.env`.

**Comment by TreLynnnn** [comment]
I did. Here's what I have. ![image](https://github.com/manaschubby/MediLink-Pro/assets/106888539/aea163bc-a226-4c5b-83ae-13f315dcae01)

**Comment by manaschubby** [comment]
Can you try finding whether Mongoose has been connected successfully? It should be logged somewhere near the start of the electron app in the terminal. Please let us know if you find it.

**Comment by manaschubby** [comment]
Also, I suggest add a `/MediLinkPro` at the end of your URI, so that your database is segregated properly.

**Comment by TreLynnnn** [comment]
Alright, I fixed the connection string and restarted the server. No error message about the db connection nor does it say it's been successfully connected. I only have this error...

--> (node:7998) UnhandledPromiseRejectionWarning: MongooseError: Operation `patients.find()` buffering timed out after 10000ms
[1]     at Timeout.<anonymous> (/Users/fcp339/Projects/MediLink-Pro/node_modules/mongoose/lib/drivers/node-mongodb-native/collection.js:185:23)
[1]     at listOnTimeout (node:internal/timers:569:17)
[1]     at process.processTimers (node:internal/timers:512:7)

**Comment by manaschubby** [comment]
This is most likely due to a failed or problematic connection to MongoDB.
Can you check whether schemas were made in your Database? Also, check whether you have added your IP address in the Database Access section of MongoDB Atlas.

**Comment by manaschubby** [comment]
![Screenshot 2023-07-07 at 9 00 47 AM](https://github.com/manaschubby/MediLink-Pro/assets/37976250/10c839eb-6b86-4db4-a7f6-b6b642d82032)

A message like this `Connected to MongoDB Atlas` should be displayed. You don't need to use Atlas. You can also use your Local Mongo for the same.

**Comment by TreLynnnn** [comment]
Okay I'll try that now

**Comment by manaschubby** [comment]
Were you successful?

**Comment by manaschubby** [comment]
Hi @TreLynnnn, if you are facing any issues please let us know!

**Comment by TreLynnnn** [comment]
Still working through it. I've already checked my cluster, updated the password and connection string. I got a failed to connect to DB message earlier, but fixed it. Still getting the same error from above. I'll work through it a bit more and see if I can make some progress. I'll update you all in a few hours. Thanks so much for your patience!

**Comment by TreLynnnn** [comment]
Finally got it fixed! 🤗 I can start working out the issue now.

**Comment by manaschubby** [comment]
That's awesome. I am glad you were able to resolve the problems. Do let us know if we can assist in any way.

**Comment by manaschubby** [comment]
Hi @TreLynnnn 
Were you able to make progress in regards to the issue?
